pkg/local_object_storage/metabase/status.go

```diff
@@ -0,0 +1,69 @@
+package meta
+
+import (
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/blobovnicza"
+	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
+	"go.etcd.io/bbolt"
+)
+
+type ObjectStatus struct {
+	State     []string
+	Path      string
+	StorageID string
+	Error     error
+}
+
+func (db *DB) ObjectStatus(address oid.Address) (ObjectStatus, error) {
+	db.modeMtx.RLock()
+	defer db.modeMtx.RUnlock()
+
+	var res ObjectStatus
+	var err error
+
+	if db.mode.NoMetabase() {
+		res.Error = ErrDegradedMode
+		return res, ErrDegradedMode
+	}
+
+	res.Path = db.boltDB.Path()
+
+	storageID := StorageIDPrm{}
+	storageID.SetAddress(address)
+	resStorageID, err := db.StorageID(storageID)
+	if id := resStorageID.StorageID(); id != nil {
+		res.StorageID = blobovnicza.NewIDFromBytes(id).String()
+	}
+
+	err = db.boltDB.View(func(tx *bbolt.Tx) error {
+		oID := address.Object()
+		cID := address.Container()
+		objKey := objectKey(address.Object(), make([]byte, objectKeySize))
+		key := make([]byte, bucketKeySize)
+
+		if objectLocked(tx, cID, oID) {
+			res.State = append(res.State, "LOCKED")
+		}
+		if inBucket(tx, primaryBucketName(cID, key), objKey) || inBucket(tx, parentBucketName(cID, key), objKey) {
+			res.State = append(res.State, "AVAILABLE")
+		}
+
+		graveyardBkt := tx.Bucket(graveyardBucketName)
+		garbageBkt := tx.Bucket(garbageBucketName)
+		addrKey := addressKey(address, make([]byte, addressKeySize))
+
+		removedStatus := inGraveyardWithKey(addrKey, graveyardBkt, garbageBkt)
+		if removedStatus != 0 && objectLocked(tx, cID, oID) {
+			res.State = append(res.State, "AVAILABLE")
+		}
+		if removedStatus == 1 {
+			res.State = append(res.State, "GC MARKED")
```

Comment:
so in the worst scenario, could it print `LOCKED AVAILABLE AVAILABLE GC MARKED`?

Reply:
it could be such a situation when an object is [inBucket](https://github.com/nspcc-dev/neofs-node/blob/1968c369653f76548bdfc05a54b71031ad6b1a07/pkg/local_object_storage/metabase/exists.go#L175 )  and [inGraveyardWithKey](https://github.com/nspcc-dev/neofs-node/blob/1968c369653f76548bdfc05a54b71031ad6b1a07/pkg/local_object_storage/metabase/exists.go#L145)?

if so it could be LOCKED AVAILABLE AVAILABLE GC MARKED